source/samples/pubsub/PubSubFeature.cpp

```diff
@@ -40,8 +92,26 @@ int PubSubFeature::init(
     thingName = *config.thingName;
     pubTopic = config.pubSub.publishTopic.value();
     subTopic = config.pubSub.subscribeTopic.value();
-    pubFile = config.pubSub.publishFile.has_value() ? config.pubSub.publishFile.value() : "";
-    subFile = config.pubSub.subscribeFile.has_value() ? config.pubSub.subscribeFile.value() : "";
+
+    if (!config.pubSub.publishFile->empty())
+    {
+        pubFile = config.pubSub.publishFile->c_str();
```

Comment:
Both `Config::PubSub::publishFile` and `Config::PubSub::subscribeFile` are of type `Aws::Crt::Optional<std::string>`. This means you need 2 tests:
* `has_value()` to confirm that the optional value is set
* `empty()` on the value to confirm that the value (`std::string`) is not empty

I won't repeat but same thing applies a few lines below for the `subscribeFile`.

Reply:
done